packages/oauth/src/state-stores/spec-utils.ts

```diff
@@ -53,11 +53,18 @@ export class StateStoreChaiTestRunner {
         it('should detect multiple consumption', async () => {
           const { stateStore } = this;
           const installUrlOptions = { scopes: ['channels:read'] };
-          for (let i = 0; i < 200; i++) {
+          for (let i = 0; i < 100; i++) {
             // generate other states
-            await stateStore.generateStateParam(installUrlOptions, new Date());
+            const date = new Date();
+            await stateStore.generateStateParam(installUrlOptions, date);
+            console.log('\tgenerateStateParam:', i, 100, date);
           }
           const state = await stateStore.generateStateParam(installUrlOptions, new Date());
+
+          // NOTE: wait 0.5 second to ensure IO operations completed
+          await new Promise((resolve, _) => setTimeout(resolve, 500));
```

Comment:
note: The comment is helpful! It may be worth clarify that we're waiting 0.5s to prevent flaky test results from rapid I/O.
```suggestion
          // Wait 0.5 second to ensure I/O is complete and avoid flaky test results from rapid I/O
          await new Promise((resolve, _) => setTimeout(resolve, 500));
```